src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs

```diff
@@ -0,0 +1,286 @@
+            }
+
+            if (typeDec.SyntaxTree.Options is CSharpParseOptions { LanguageVersion: < LanguageVersion.CSharp10 })
+            {
+                return Diagnostic.Create(DiagnosticDescriptors.InvalidLangVersion, methodSyntax.GetLocation());
+            }
+
+            regexMethod.MethodName = regexMethodSymbol.Name;
+            regexMethod.Modifiers = methodSyntax.Modifiers.ToString();
+            regexMethod.MatchTimeout ??= Timeout.Infinite;
+            RegexOptions options = regexMethod.Options.HasValue ? (RegexOptions)regexMethod.Options.Value : RegexOptions.None;
+            regexMethod.Options = (int)RegexOptions.Compiled | (int)options;
+
+            // TODO: This is going to include the culture that's current at the time of compilation.
+            // What should we do about that?  We could:
+            // - say not specifying CultureInvariant is invalid if anything about options or the expression will look at culture
+            // - fall back to not generating source if it's not specified
+            // - just use whatever culture is present at build time
+            // - devise a new way of not using the culture present at build time
+            // - ...
+            CultureInfo culture = (options & RegexOptions.CultureInvariant) != 0 ? CultureInfo.InvariantCulture : CultureInfo.CurrentCulture;
+
+            // Validate the options
+            const RegexOptions SupportedOptions =
+                RegexOptions.IgnoreCase |
+                RegexOptions.Multiline |
+                RegexOptions.ExplicitCapture |
+                RegexOptions.Compiled |
+                RegexOptions.Singleline |
+                RegexOptions.IgnorePatternWhitespace |
+                RegexOptions.RightToLeft |
+                RegexOptions.ECMAScript |
+                RegexOptions.CultureInvariant;
+            if ((regexMethod.Options.Value & ~(int)SupportedOptions) != 0)
+            {
+                return Diagnostic.Create(DiagnosticDescriptors.InvalidRegexArguments, methodSyntax.GetLocation(), "options");
+            }
+
+            // Validate the timeout
+            if (regexMethod.MatchTimeout.Value is 0 or < -1)
+            {
+                return Diagnostic.Create(DiagnosticDescriptors.InvalidRegexArguments, methodSyntax.GetLocation(), "matchTimeout");
+            }
+
+            // Parse the input pattern
+            try
+            {
+                regexMethod.Tree = RegexParser.Parse(regexMethod.Pattern, (RegexOptions)regexMethod.Options, culture);
+                regexMethod.Code = RegexWriter.Write(regexMethod.Tree);
+            }
+            catch (Exception e)
+            {
+                return Diagnostic.Create(DiagnosticDescriptors.InvalidRegexArguments, methodSyntax.GetLocation(), e.Message);
+            }
+
+            // Determine the namespace the class is declared in, if any
```

Comment:
This is one place in the Logging Generator I'd like changed. The Json generator does it better. Instead of interpreting the parents (and potentially having bugs when new features, like file scoped namespaces, are created), I think it is better to get the TypeSymbol of the containing type, and get its namespace that way. This is how the Json generator does it.

See https://github.com/dotnet/runtime/pull/57894#issuecomment-918247048